tests/base.py

```diff
@@ -76,7 +77,9 @@ def expected_metadata(self):
             "conversation_threads": {
                 self.PRIMARY_KEYS: {"id"},
                 self.REPLICATION_METHOD: self.FULL_TABLE,
-                self.EXPECTED_PAGE_SIZE: 50
+                self.EXPECTED_PAGE_SIZE: 50,
+                self.FOREIGN_KEYS: {"conversation_id"},
+                self.EXPECTED_PARENT_STREAM: "conversations"
```

Comment:
Use the OBEY_START_DATE key here to define if the stream is following start_date or not, and use this in the start_date test. I got this comment in the past, so following it for all the taps now.